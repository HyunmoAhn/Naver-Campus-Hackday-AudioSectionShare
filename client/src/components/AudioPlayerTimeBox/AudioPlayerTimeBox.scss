.AudioPlayerTimeBox {
  width: 100%;
  overflow: hidden;
}

.AudioPlayerTimeBox__range {
  display: inline-block;
  margin: auto;
  width: 70%;
  text-align: center;
  vertical-align: middle;
}

