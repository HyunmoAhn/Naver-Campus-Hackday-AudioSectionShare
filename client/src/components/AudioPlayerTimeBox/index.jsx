import React from 'react';
import PropTypes from 'prop-types';
import TimeScreen from '../TimeScreen';
import './AudioPlayerTimeBox.scss';

const propTypes = {
	currentTime: PropTypes.number,
	duration: PropTypes.number,
	onCurrentTimeChange: PropTypes.func,
};

const defaultProps = {
	currentTime: 0,
	duration: 0,
	onCurrentTimeChange() {},
};

class AudioPlayerTimeBox extends React.Component {
	render() {
		const { currentTime, duration, onCurrentTimeChange } = this.props;
		const relativeCurrentTime = currentTime / duration * 100;

		return (
			<div className="AudioPlayerTimeBox">
				<TimeScreen className="AudioPlayerTimeBox__currentTime" second={currentTime} />
				<div
					className="AudioPlayerTimeBox__range"
				>
					<input
						type="range"
						value={relativeCurrentTime}
						onChange={onCurrentTimeChange}
					/>
				</div>
				<TimeScreen className="AudioPlayerTimeBox__duration" second={duration} />
			</div>
		);
	}
}

AudioPlayerTimeBox.propTypes = propTypes;
AudioPlayerTimeBox.defaultProps = defaultProps;

export default AudioPlayerTimeBox;