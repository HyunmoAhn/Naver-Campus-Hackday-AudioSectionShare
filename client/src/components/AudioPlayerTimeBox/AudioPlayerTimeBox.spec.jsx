import React from 'react';
import { shallow } from 'enzyme';

import AudioPlayerTimeBox from './';

describe('<AudioPlayerTimeBox />', () => {
	it('should match to snapshot when render default', () => {
		const wrapper = shallow(<AudioPlayerTimeBox />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should call onCurrentTImeChange when ca', () => {
		const onCurrentTimeChange = jest.fn();
		const wrapper = shallow(<AudioPlayerTimeBox
			onCurrentTimeChange={onCurrentTimeChange}
		/>);
		const range = wrapper.find('.range');

		range.simulate('change');
		expect(onCurrentTimeChange.mock.calls.length).toBe(1);
	});
});