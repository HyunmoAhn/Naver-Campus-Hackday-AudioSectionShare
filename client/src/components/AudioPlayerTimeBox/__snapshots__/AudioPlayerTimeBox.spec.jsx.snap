// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioPlayerTimeBox /> should match to snapshot when render default 1`] = `
<div
  className="AudioPlayerTimeBox"
>
  <TimeScreen
    className="AudioPlayerTimeBox__currentTime"
    second={0}
  />
  <div
    className="AudioPlayerTimeBox__range range__container"
  >
    <input
      className="AudioPlayerTimeBox__range-input range"
      onChange={[Function]}
      type="range"
      value={NaN}
    />
  </div>
  <TimeScreen
    className="AudioPlayerTimeBox__duration"
    second={0}
  />
</div>
`;
